tools/cwrap/plugins/Broadcast.py

```diff
@@ -75,29 +90,44 @@ def getExpandTemplate(self, expand_call, success_code, raise_errors):
                 "catch (std::exception &e) {}\n" +
                 "if(expand_success) {\n" +
                 success_code +
+                "\n}" +
                 "\n}\n")
         else:
             return Template(
+                "bool try_expand = !" + same_size_check + "\n" +
+                "if (try_expand) {\n" +
                 expand_call + "\n" +
-                success_code + "\n")
-
-    def getOutPlacePreExpand2Template(self, raise_errors):
-        expand_code = """expand_outplace2(LIBRARY_STATE ${arg_op_a}_guard.get(), ${arg_op_other}_guard.get(),
-                                          ${arg_op_a}, ${arg_op_other},
-                                          \"${op_a}\", \"${op_other}\", !${raise_errors});"""
+                success_code + "\n"
+                "}\n")
+
+    def getOutPlacePreExpand2Template(self, type_op_a, type_op_other, raise_errors):
+        size_check = """THSize_isSameSizeAs(${arg_op_a}->size, ${arg_op_a}->nDimension,
```

Comment:
It's not necessary; this is just checking if we can early-opt out of the broadcasting code, so it works as if the broadcasting code wasn't there.  So, assuming these operations worked on tensors of same size but different strides, this code should work as well.